util/utils.go

```diff
@@ -1,49 +1,49 @@
 package util
 
 import (
+	"log"
+	"os"
 	"os/exec"
-	"time"
-	"strings"
-	"fmt"
 	"os/signal"
-	"os"
+	"strings"
 	"syscall"
+	"time"
 )
 
 type Change struct {
-	BlockId int
-	Data string
+	BlockID int
+	Data    string
 	Success bool
 }
+
 //Based on "timer" prorty from config file
-//Schedule gothread that will ping other gothreads via send channel
-func Schedule(send chan bool, duration string){
+//Schedule goroutine that will ping other goroutines via send channel
+func Schedule(send chan bool, duration string) {
 	u, err := time.ParseDuration(duration)
-	if err == nil {
-		for {
-			send <- true
-			time.Sleep(u)
-		}
-	} else {
-		fmt.Println("Couldn't set a scheduler due to improper time format: "+ duration)
+	if err != nil {
+		log.Fatalf("Couldn't set a scheduler due to improper time format: %s\n", duration)
+	}
+	for range time.Tick(u) {
+		send <- true
 	}
 }
+
 //Run any arbitrary POSIX shell command
-func RunCmd(blockId int, send chan Change, rec chan bool, action map[string]interface{} ) {
+func RunCmd(blockID int, send chan Change, rec chan bool, action map[string]interface{}) {
 	cmdStr := action["command"].(string)
```

Comment:
This for example. What if command is not a key? And you have to extract the underlying type (which is ugly). Precisely why we have structs

Reply:
Yeah I guess I should check for "command" and "timer" parameters. As to the awkward handling of actions I explained it in previous review.